br/pkg/lightning/backend/external/util.go

```diff
@@ -186,3 +187,50 @@ func MockExternalEngine(
 	}
 	return GetAllFileNames(ctx, storage, "/mock-test")
 }
+
+// EndpointTp is the type of Endpoint.Key.
+type EndpointTp int
+
+const (
+	// ExclusiveEnd represents "..., Endpoint.Key)".
+	ExclusiveEnd EndpointTp = iota
```

Comment:
I guess we don't need it?

Reply:
Yes currently we don't need it. But I'm not sure there will be usage in future like "checking the overlap of RangeSplitter" or something.